File: tests/fixtures/cpp/basic_pkg/src/dog.cpp
#include "dog.h"

#include <utility>

namespace acme {

Dog::Dog(std::string name) : name_(std::move(name)) {}

std::string Dog::speak() const {
    Sound s("woof");
    Sound louder = s.amplify();
    return name_ + " says " + louder.text();
}

std::string Dog::name() const { return name_; }

int Dog::bumpAge(int current, int delta) {
    return current + delta;
}

}  // namespace acme

File: tests/fixtures/objc/basic_pkg/Animals/Dog.m
#import "Dog.h"

@implementation Dog
{
    NSString *_name;
}

- (instancetype)initWithName:(NSString *)name {
    self = [super init];
    if (self) {
        _name = [name copy];
    }
    return self;
}

- (NSString *)speak {
    Sound *s = [[Sound alloc] initWithText:@"woof"];
    Sound *louder = [s amplify];
    return [NSString stringWithFormat:@"%@ says %@", _name, [louder text]];
}

- (NSString *)name {
    return _name;
}

- (NSInteger)bumpAge:(NSInteger)current by:(NSInteger)delta {
    return current + delta;
}

- (id)copyWithZone:(NSZone *)zone {
    return [[Dog allocWithZone:zone] initWithName:_name];
}
@end
